Move Extension Localization Information from Extbase Fluid Book (#2065)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ExtensionArchitecture/HowTo/Localization/Fluid.rst

.. include:: /Includes.rst.txt
.. index::
   Fluid Templates; Multi-language
   Localization; Fluid templates
.. _extension-localization-fluid:

==============================
Multi-language Fluid templates
==============================

.. tip::

>    Texts can also be translated direclty in the PHP code, for example in

Reviewer: ```suggestion
   Texts can also be translated directly in the PHP code, for example in
```

   <h3>Comments</h3>
   <f:for each="{post.comments}" as="comment">
     {comment.content -> f:format.nl2br()}
     <hr>
   </f:for>

To make such texts exchangeable, they have to be removed from the Fluid
template and inserted into a
:ref:`language file <extension-localization-language-file>`. Every text
fragment which is to be translated is given an identifier (also called key)
that can be inserted in the Fluid template.

.. index::
   pair: Fluid; Localization
   Fluid; f:translate

.. _f-translate:

The translation ViewHelper :html:`f:translate`
==============================================

To insert translations into a template, Fluid offers the ViewHelper
:ref:`f:translate <t3viewhelper:typo3-fluid-translate>`.

This ViewHelper has a property called :html:`key` where the identifier of
the text fragment prefixed by the location file can be provided.

.. code-block:: html
   :caption: EXT:my_extension/Resources/Private/Templates/SomeTemplate.html

   <f:translate key="LLL:EXT:my_extension/Resources/Private/Language/yourFile.xlf:yourKey" />
   <!-- or as inline Fluid: -->
   {f:translate(key: 'LLL:EXT:my_extension/Resources/Private/Language/yourFile.xlf:yourKey')}

> The text fragment will now me output in the current frontend language

Reviewer: ```suggestion
The text fragment will now be displayed in the current frontend language
```

> defined in the site configuration if the translation file of the requested

Reviewer: ```suggestion
defined in the site configuration, if the translation file of the requested
```

> langauge can be found in the location of the prefix.

Reviewer: ```suggestion
language can be found in the location of the prefix.
```
